internal/shared/services/allow_list.go

```diff
@@ -0,0 +1,64 @@
+package services
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/babylonlabs-io/staking-api-service/internal/shared/config"
+	"github.com/rs/zerolog/log"
+)
+
+func loadAllowList(cfg *config.Config) (map[string]bool, error) {
+	if cfg.AllowList == nil || cfg.AllowList.FilePath == "" {
```

Comment:
I think if AllowList is set it's a mistake to keep FilePath empty i.e. we should add validation to `AllowList` and remove check that it's empty here